datajunction-server/datajunction_server/api/nodes.py

```diff
@@ -910,10 +910,130 @@ async def link_dimension(
     )
 
 
+@router.post("/nodes/{node_name}/columns/{node_column}/link", status_code=201)
+async def add_reference_dimension_link(
+    node_name: str,
+    node_column: str,
+    dimension_node: str,
+    dimension_column: str,
+    role: Optional[str] = None,
+    session: AsyncSession = Depends(get_session),
+    current_user: User = Depends(get_and_update_current_user),
+) -> JSONResponse:
+    """
+    Add reference dimension link to a node column
+    """
+    node = await Node.get_by_name(session, node_name, raise_if_not_exists=True)
+    dim_node = await Node.get_by_name(session, dimension_node, raise_if_not_exists=True)
+    if dim_node.type != NodeType.DIMENSION:  # type: ignore
+        raise DJInvalidInputException(
+            message=f"Node {node.name} is not of type dimension!",  # type: ignore
+        )
+
+    # The target and dimension columns should both exist
+    target_column = await get_column(session, node.current, node_column)  # type: ignore
+    dim_column = await get_column(session, dim_node.current, dimension_column)  # type: ignore
```

Comment:
This may be unnecessary, but I am wondering if we should also make sure that the `dim_column` is not the only PK of this dim node. If we don't check it, we are risking that the user may also create a regular dimension link from this column to the same or different dim node. It's a bit of an F&R space but there is a risk of redundancy and design mismatches if we allow PK columns here imo.

Reply:
Hmm yeah, this is probably a good idea. In general I think having more restrictions for users will be better in the process of creating the dimensions graph. However, I'm going to punt this to a separate PR